tools/userland-component

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/python3.5
+
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2021 Aurelien Larcher
+#
+
+import argparse
+import os
+import re
+import sys
+import json
+
+from bass.component import Component
+from bass.makefiles import Item
+from bass.makefiles import Keywords
+from bass.makefiles import Makefile as MK
+
+# Refactoring rules
+# 000:  Use WS_* variables instead $(WS_TOP)/* 
+#       If $(WS_TOP)/make-rules is found in an include then replace with the
+#       variable $(WS_RULES). Do the same for other variables.
+def refactor000(mk):
+    for i in iter(mk.includes):
+        r = re.match(r"^\$\(WS_TOP\)\/(.*)\/(.*).mk", i.str[0])
+        if r is not None:
+            subdir = r.group(1)
+            mkfile = r.group(2)
+            print("000: Fix include " + i.str[0])
+            i.str = [MK.value("WS_"+subdir.upper().replace("-","_"))+"/"+mkfile+".mk"]
+            mk.contents[i.idx] = i.include_line()
+
+
+# 001:  Use common.mk
+#       If common.mk is not included then:
+#           1. infer the build system and set the BUILD_STYLE.
+#           2. set the BUILD_BITS from the existing targets.
+#           3. erase default target and keep the custom ones.
+def refactor001(mk):
+    kw = Keywords()
```

Comment:
refactor001() -> refactor_use_common_mk()

Reply:
And refactor some parts of its body into separate helper methods. Maybe some of them can be re-used by other refactor_* methods later.